pkg/scheduler/plugins/deviceshare/plugin.go

```diff
@@ -72,25 +72,29 @@ var (
 )
 
 type Plugin struct {
-	disableDeviceNUMATopologyAlignment bool
-	handle                             frameworkext.ExtendedHandle
-	nodeDeviceCache                    *nodeDeviceCache
-	scorer                             *resourceAllocationScorer
+	disableDeviceNUMATopologyAlignment         bool
+	handle                                     frameworkext.ExtendedHandle
+	nodeDeviceCache                            *nodeDeviceCache
+	gpuSharedResourceTemplatesCache            *gpuSharedResourceTemplatesCache
+	gpuSharedResourceTemplatesMatchedResources []corev1.ResourceName
+	scorer                                     *resourceAllocationScorer
 }
 
 type preFilterState struct {
-	skip                              bool
-	podRequests                       map[schedulingv1alpha1.DeviceType]corev1.ResourceList
-	hints                             apiext.DeviceAllocateHints
-	hintSelectors                     map[schedulingv1alpha1.DeviceType][2]labels.Selector
-	hasSelectors                      bool
-	jointAllocate                     *apiext.DeviceJointAllocate
-	primaryDeviceType                 schedulingv1alpha1.DeviceType
-	podFitsSecondaryDeviceWellPlanned bool
-	gpuRequirements                   *GPURequirements
-	allocationResult                  apiext.DeviceAllocations
-	preemptibleDevices                map[string]map[schedulingv1alpha1.DeviceType]deviceResources
-	preemptibleInRRs                  map[string]map[types.UID]map[schedulingv1alpha1.DeviceType]deviceResources
+	skip                                bool
+	podRequests                         map[schedulingv1alpha1.DeviceType]corev1.ResourceList
+	hints                               apiext.DeviceAllocateHints
+	hintSelectors                       map[schedulingv1alpha1.DeviceType][2]labels.Selector
+	hasSelectors                        bool
+	jointAllocate                       *apiext.DeviceJointAllocate
+	primaryDeviceType                   schedulingv1alpha1.DeviceType
+	podFitsSecondaryDeviceWellPlanned   bool
+	gpuRequirements                     *GPURequirements
+	enforceGPUSharedResourceTemplate    bool
```

Comment:
Can these two parameters be put into preFilterState.GPURequirements? If not, then the preFilterState.clone function needs to clone these two parameters as well.